File: src/main/java/com/dinosaur/dinosaurexploder/constants/Direction.java
/*
 * SPDX-FileCopyrightText: 2026 jvondermarck
 * SPDX-License-Identifier: MIT
 */

package com.dinosaur.dinosaurexploder.constants;

import static com.almasb.fxgl.core.math.FXGLMath.random;

public enum Direction {
  UP,
  DOWN,
  LEFT,
  RIGHT;

  public static Direction randomDirection() {
    Direction[] values = Direction.values();
    int index = random(0, values.length - 1);
    return values[index];
  }
}

File: src/main/kotlin/com/dinosaur/dinosaurexploder/constants/Direction.kt
/*
 * SPDX-FileCopyrightText: 2026 jvondermarck
 * SPDX-License-Identifier: MIT
 */
package com.dinosaur.dinosaurexploder.constants

import com.almasb.fxgl.core.math.FXGLMath.random

enum class Direction {
    UP, DOWN, LEFT, RIGHT;

    companion object {
        @JvmStatic
        fun modeDirection(mode: GameMode): Direction {
            return if (mode == GameMode.EXPERT) randomDirection() else UP
        }

        @JvmStatic
        fun randomDirection(): Direction {
            val values = values()
            val index = random(0, values.size - 1)
            return values[index]
        }
    }
}